### ISSUE TITLE
NullPointerException when fetching Entity with a null-valued @Embedded property

### ISSUE THREAD

**Issue by gbrancovici-sam** [Impetus/kundera#774]
We have an entity class with @Embedded field and its corresponding class is marked with @Embeddable. We encountered problems in fetching entities whenever this embedded field is null. Following the stack trace for your reference

``` java
! java.lang.NullPointerException: null
! at com.datastore.kundera.client.cassandra.dsdriver.DSClientUtilities.setBasicValue(DSClientUtilities.java:521) ~[kundera-client-cassandra-ssl-0.0.2-SNAPSHOT.jar:na]
! at com.datastore.kundera.client.cassandra.dsdriver.DSClientUtilities.setUDTValue(DSClientUtilities.java:429) ~[kundera-client-cassandra-ssl-0.0.2-SNAPSHOT.jar:na]
! at com.datastore.kundera.client.cassandra.dsdriver.DSClientUtilities.assign(DSClientUtilities.java:386) ~[kundera-client-cassandra-ssl-0.0.2-SNAPSHOT.jar:na]
! at com.datastore.kundera.client.cassandra.dsdriver.DSClient.iteratorColumns(DSClient.java:898) ~[kundera-client-cassandra-ssl-0.0.2-SNAPSHOT.jar:na]
! at com.datastore.kundera.client.cassandra.dsdriver.DSClient.iterateAndReturn(DSClient.java:790) ~[kundera-client-cassandra-ssl-0.0.2-SNAPSHOT.jar:na]
! at com.datastore.kundera.client.cassandra.dsdriver.DSClient.find(DSClient.java:173) ~[kundera-client-cassandra-ssl-0.0.2-SNAPSHOT.jar:na]
! at com.impetus.kundera.persistence.AbstractEntityReader.findById(AbstractEntityReader.java:81) ~[kundera-core-3.0.jar:na]
! ... 66 common frames omitted
! Causing: com.impetus.kundera.persistence.EntityReaderException: java.lang.NullPointerException
! at com.impetus.kundera.persistence.AbstractEntityReader.findById(AbstractEntityReader.java:95) ~[kundera-core-3.0.jar:na]
! at com.impetus.client.cassandra.query.CassandraEntityReader.findById(CassandraEntityReader.java:93) ~[kundera-cassandra-3.0.jar:na]
! at com.impetus.kundera.lifecycle.states.ManagedState.handleFind(ManagedState.java:137) ~[kundera-core-3.0.jar:na]
! at com.impetus.kundera.graph.Node.find(Node.java:500) ~[kundera-core-3.0.jar:na]
! at com.impetus.kundera.persistence.PersistenceDelegator.find(PersistenceDelegator.java:225) ~[kundera-core-3.0.jar:na]
! at com.impetus.kundera.persistence.PersistenceDelegator.findById(PersistenceDelegator.java:174) ~[kundera-core-3.0.jar:na]
! at com.impetus.kundera.persistence.EntityManagerImpl.find(EntityManagerImpl.java:263) ~[kundera-core-3.0.jar:na]
```

**Comment by amit-kr** [comment]
@haidukus 

Fix has been added in the trunk. Please build it and let us know.

-Amit

**Comment by gbrancovici-sam** [comment]
Thanks, I will!

**Comment by gbrancovici-sam** [comment]
Hi!

The fix seems to work fine so far.

Another question: I don't have the chance at this moment to follow through the code properly and do tests, but do you think that we should also check whether 'member' is null, e.g. do something like:

```
        case UDT:
            retVal = row.getUDTValue(columnName);
            if (member != null) {
                    PropertyAccessorHelper.set(entity, member,
                    retVal != null ? setUDTValue(entity, member.getType(), (UDTValue) retVal, metamodel) : null);
            }
            break;
```

**Comment by amit-kr** [comment]
Hi @haidukus 

Thanks for pointing out, null check for member should be there. We have fixed the issue and is available in current trunk.

-Amit

**Comment by gbrancovici-sam** [comment]
Cool!

In that case we might want to update the following as well:
`case CUSTOM:
            retVal = row.getBytes(columnName);
            if (member != null && retVal != null && entity != null)
            {
                PropertyAccessorHelper.set(entity, member, ((ByteBuffer) retVal).array());
            }
            break;`
... to using setFieldValue :)

**Comment by chhavigangwal** [comment]
@haidukus 

This piece of code will require an explicit check for retVal to be not null before type casting it to ByteBuffer, so shouldn't really make a difference if changed to setFieldValue (which will add a second null ) or not .

Chhavi

**Comment by gbrancovici-sam** [comment]
Got it, thanks!

**Comment by chhavigangwal** [comment]
Resolved , closing with Kundera 3.1 release
